src/repl-frontend/scala/tools/nsc/interpreter/shell/ShellConfig.scala

```diff
@@ -38,7 +38,7 @@ object ShellConfig {
       val batchText: String = if (settings.execute.isSetByUser) settings.execute.value else ""
       val batchMode: Boolean = batchText.nonEmpty
       val doCompletion: Boolean = !(settings.noCompletion.value || batchMode)
-      override val haveInteractiveConsole: Boolean = super.haveInteractiveConsole && !settings.Xnojline.value
+      def haveInteractiveConsole: Boolean = consoleIsTerminal && !settings.Xnojline.value
```

Comment:
Can you tell me why did we change `haveInteractiveConsole` in the first place? Was something broken?

Reply:
I had the same question last night. It crossed my field of vision. There is another ticket to remove `scala -Xjline`. Oh, my first tweak was here instead of Properties.

How about I reset to previous instead of raising dust?